From 1c40af5de0ac8ac6670f417db06b5cee8233191c Mon Sep 17 00:00:00 2001
From: Caesar Wang <wxt@rock-chips.com>
Date: Wed, 25 Apr 2018 18:31:38 +0800
Subject: [PATCH] arm64: dts: rockchip: rk3326-evb-linux-lp3-v10: add the
 ov5695 for rk3326 mipi

This patch adds the ov5695 releated configure with rk3326 mipi/isp.

Change-Id: I62d6d8854d7e66fac521b3fea0f4dd35dc2799c4
Signed-off-by: Caesar Wang <wxt@rock-chips.com>
---
 .../dts/rockchip/rk3326-evb-linux-lp3-v10.dts | 81 +++++++++++++++++--
 1 file changed, 76 insertions(+), 5 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3326-evb-linux-lp3-v10.dts b/arch/arm64/boot/dts/rockchip/rk3326-evb-linux-lp3-v10.dts
index 2c5f6eeb52b4..1f365619eaf9 100644
--- a/arch/arm64/boot/dts/rockchip/rk3326-evb-linux-lp3-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3326-evb-linux-lp3-v10.dts
@@ -687,6 +687,35 @@
 
 &i2c2 {
 	status = "okay";
+	clock-frequency = <400000>;
+
+	/* These are relatively safe rise/fall times; TODO: measure */
+	i2c-scl-falling-time-ns = <50>;
+	i2c-scl-rising-time-ns = <300>;
+
+	ov5695: ov5695@36 {
+		compatible = "ovti,ov5695";
+		reg = <0x36>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&cif_clkout_m0>;
+
+		clocks = <&cru SCLK_CIF_OUT>;
+		clock-names = "xvclk";
+
+		avdd-supply = <&vcc2v8_dvp>;
+		dovdd-supply = <&vcc1v8_dvp>;
+		dvdd-supply = <&vcc1v8_dvp>;
+
+		/*reset-gpios = <&gpio2 14 GPIO_ACTIVE_HIGH>;*/
+		pwdn-gpios = <&gpio2 14 GPIO_ACTIVE_HIGH>;
+
+		port {
+			ucam_out: endpoint {
+				remote-endpoint = <&mipi_in_ucam>;
+				data-lanes = <1 2>;
+			};
+		};
+	};
 };
 
 &i2s1_2ch {
@@ -700,7 +729,7 @@
 	vccio1-supply = <&vcc1v8_soc>;
 	vccio2-supply = <&vccio_sd>;
 	vccio3-supply = <&vcc1v8_dvp>;
-	vccio4-supply = <&vcc_3v0>;
+	vccio4-supply = <&vcc1v8_soc>;
 	vccio5-supply = <&vcc_3v0>;
 };
 
@@ -730,10 +759,56 @@
 	status = "okay";
 };
 
+&mipi_dphy_rx0 {
+	status = "okay";
+
+	ports {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		port@0 {
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			mipi_in_ucam: endpoint@1 {
+				reg = <1>;
+				remote-endpoint = <&ucam_out>;
+				data-lanes = <1 2>;
+			};
+		};
+
+		port@1 {
+			reg = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			dphy_rx0_out: endpoint@0 {
+				reg = <0>;
+				remote-endpoint = <&isp0_mipi_in>;
+			};
+		};
+	};
+};
+
 &nandc0 {
 	status = "okay";
 };
 
+&rkisp1 {
+	status = "okay";
+
+	port {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		isp0_mipi_in: endpoint@0 {
+			reg = <0>;
+			remote-endpoint = <&dphy_rx0_out>;
+		};
+	};
+};
+
 &pinctrl {
 	headphone {
 		hp_det: hp-det {
@@ -781,10 +856,6 @@
 	status = "okay";
 };
 
-&rk_isp {
-	status = "okay";
-};
-
 &rk_rga {
 	status = "okay";
 };
-- 
2.35.3

